clouddq/classes/rule_type.py

```diff
@@ -84,6 +84,26 @@ def to_sql_custom_sql_expr(params: dict) -> Template:
         f"'custom_sql_expr' in 'params'.\n"
         f"Current value: {custom_sql_expr}",
     )
+    custom_sql_arguments = params.get("custom_sql_arguments", "")
+    if custom_sql_arguments:
+        for argument in custom_sql_arguments:
+            if f"${argument}" not in custom_sql_expr:
+                raise ValueError(
```

Comment:
Should this be error or warning? Not used parameters might indicate something is not as intended but not necessarily be a fatal problem (similar to non-used function parameters in mainstream languages)

Reply:
It's an interesting question. I think if no error is thrown (but rather warning) there's risk of leaving behind a hardcoded parameter. It can happen that while developing rules (and testing them directly in the query engine) and then passing them to the yaml file, there's a risk of keeping the hardcoded parameters, so a failure (e.g. when tests) might be a desirable. But I can see the other point of, not being so conservative and letting the execution go.

In any case, the same should apply to CUSTOM_SQL_EXPRESSION.

Reply:
I would rather be strict here to avoid confusions. We can easily relax restrictions in the future, but we cannot as easily add restrictions that did not exist before.

When we look at adding curried parameters we can relax this.

Reply:
This works for me.

Reply:
agree we can relax later as a non-breaking change so let's go ahead with a warning here 👍